remove deprecated or dead code

jonsneyers opened the pull request:
Remove the deprecated DCOutBuffer stuff from the decode API — this was deprecated a long time ago and didn't work anymore anyway, so no need to keep it in.

Also remove some code that is no longer used. Move the speed tier name parsing to benchmark_xl (which is the only thing that still uses that). The function to convert speed settings to animal name strings was only used to produce pretty names for the tests in `speed_tier_test.cc`, so removing that too. Some overloads of `UnpremultiplyAlpha` and `ConvertToExternal` are not used by anything anymore, so removing them too.

szabadka commented on lib/include/jxl/decode.h:
```diff
@@ -263,15 +253,10 @@ typedef enum {
+   * it does it will do so before outputting the full frame. This return status
+   * only indicates we're past this point in the codestream. This event occurs
+   * max once per frame and always later than @ref JXL_DEC_FRAME and other
+   * header events and earlier than full resolution pixel data.
```
We should say here that we never return this event any more.

jonsneyers replied:
Maybe better to just remove this from the enum. If an application is still subscribing to this event, it's probably better to give them a compile error so they can update their code to do progressive rendering correctly, than it is to keep things 'working' but never actually produce the event...

szabadka replied:
Yes, I agree, it is better to remove it, since it is marked as deprecated anyway.

jonsneyers commented:
Updated: removed JXL_DEC_DC_IMAGE completely (it already didn't do anything for a long time, but now applications that still subscribe to it will get a compile error).

szabadka reviewed (approved):
Thanks for the cleanup!
